packages/grafana-ui/src/components/PanelChrome/PanelContext.ts

```diff
@@ -82,6 +84,17 @@ export interface PanelContext {
    * Called when a panel is changing the sort order of the legends.
    */
   onToggleLegendSort?: (sortBy: string) => void;
+
+  /**
+   * Some panels could be interested in updating the queries.
+   * This optional and not always available in all contexts.
+   */
```

Comment:
This one is odd since a panel viz does not know the queries

Reply:
True, but in the datagrid scenario it can create new snapshot queries

Reply:
@ryantxu we could add queries to PanelProps